batch.ts

```diff
@@ -0,0 +1,114 @@
+import {transact, getMutationID, setMutationID} from './db';
+import {gcal} from './gcal';
+import {userError, ErrorKind} from './errors';
+
+export async function batch(payload: any, auth: string) {
+    validatePayload(payload);
+
+    let infos = [];
+    for (let m of payload.mutations) {
+        await transact(async (db) => {
+            const currentMutationID = await getMutationID(db, payload.clientID);
+            const expectedMutationID = currentMutationID + 1;
+
+            if (m.id > expectedMutationID) {
+                throw userError(`mutation.id is too high - expected ${expectedMutationID}`);
```

Comment:
We haven't to this point really had a concept of a user error when it comes to the client View or batch end point. Talk about temporary or permanent errors. Should we be using that terminology here? I think this would be a temporary failure and should as it does here abort the batch run